### ISSUE TITLE
Multiple database install

### ISSUE THREAD

**Issue by kiaking** [vuex-orm/vuex-orm#515]
Ref #360 #514

Enable having more than 1 database installed to a single vuex store.

### Registration API

A user should be able to register more than 1 database to the vuex store by separating the namespace.

```js
import Vuex from 'vuex'
import VuexORM from '@vuex-orm/core'
import User from '@/models/User'
import Video from '@/models/Video'

const databae1 = new VuexORM.Database()
database1.register(User)

const databae2 = new VuexORM.Database()
database2.register(Video)

const store = new Vuex.Store({
  plugins: [
    VuexORM.install(database1, { namespace: 'database1' }),
    VuexORM.install(database1, { namespace: 'database2' })
  ]
})
```

### How models will be able to determine which database to access?

During the installation process, Vuex ORM will attach database name to the registered models. The model will have a new static property called `connection`. This property should hold the namespace of the database so that it can use it to retrieve the correct database instance from the Container object.

We don't have to worry about modules because, at the time when the module method is called, it knows which namespace it's getting called under.

#### Models can't have relationships with different databases.

For example, User model at database1 can't have relationships with Video model at database2. Database separation is meant for complete distinguish between different databases.

Because to be able to have relationships between different databases, we must either create modules for at the root of vuex namespace or deepen Vuex ORM namespace to something like `databases/database1/users`. I think this is too much of a work, and it's better to keep everything inside a single database for such use cases.

#### Same models can't belong in more than one databases.

For example, if you register the User model to database1, you can't register the same model to database2. This is because the model needs to know on what database it belongs to.

I don't think this is going to be a problem since I don't think you want to have the same type of records at 2 different places, it kinda eliminates the need of Vuex ORM in the first place. But if anyone needs to reuse the logic of the model, they can easily do that by creating a base model class and inherit it.

```js
class BaseUser extends Model {
  static fields () {
    return {...}
  }
}

class UserForDbA extends BaseUser {}

class UserForDbB extends BaseUser {}

database1.register(UserForDbA)
database2.register(UserForDbB)
```

**Comment by moravecjakub** [comment]
Hello @kiaking,

  currently I'm in trouble with multiple databases in one application. I'm trying to find some way how to use more databases with current implementation. I chose the way to replace your singleton container for one database with my own container, which can handle more databases:
```javascript
class DBContainer {
    constructor () {
        this.databases = {};
    }
    register (database, name) {
        this.databases[name] = database;
    }
    getDatabase (name) {
        return this.databases[name];
    }
}
export default new DBContainer(); // singleton
```
Then I'm able to make bypass for container via plugin for orm:
```javascript
import {Container} from '@vuex-orm/core';
export default {
    install ({Query, Model}, {namespace}) {
        // replace original functionality
        Container.register = function (database) {
            DBContainer.register(database, namespace);
        };
        // replace original static function
        Model.database = function () {
            return DBContainer.getDatabase(this.dbName);
        };
        // replace original static function - not work - explained below
        Query.database = function () {
            return DBContainer.getDatabase(???);
        };
    }
};
```
Which is used as by standard wayt with same namespace as for database namespace:
```javascript
VuexORM.use(NamespacePlugin, {namespace: 'resolveRegistrationDB'});
```
And finaly it needs to define property 'dbName' into each model:
```javascript
export default class Answer extends Model {
    static dbName = 'resolveRegistrationDB';
...
}
```
## The issue:
As we can see in plugin definition, I'm not able to work with database namespace for query. The problem is the static context. As we can see in Query constructor, there is a 'state' as first argument, which contains wanted namespace, which would be applicable in my method for getting DB in query - see '???' in plugin. But, this method is called in static context, why? Is it really needed to have static context in this place? So, in my implementation above is the last only one issue, how to recognize right database in query static context - which is impossible, I think.

I need to solve it very quickly, because it fails on our production :( Has anybody some idea how I can implement some temporary solution until it will be solved in this library?

**Comment by kiaking** [comment]
@moravecjakub Yeah, to bypass this, we must somehow tell Query what database to use. And to do that, I think we need to pass additional argument to the Vuex Modules...
